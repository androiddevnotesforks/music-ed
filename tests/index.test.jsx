import { render, screen } from '@testing-library/react'
import Home from '../pages/index'
import '@testing-library/jest-dom'
import HomeComponent from '../components/home/homeComponent';

jest.mock("next-auth/react", () => {
  const originalModule = jest.requireActual('next-auth/react');
  const mockSession = null;
  return {
    __esModule: true,
    ...originalModule,
    useSession: jest.fn(() => {
      return {data: mockSession, status: 'authenticated'}  // return type is [] in v3 but changed to {} in v4
    }),
  };
});


describe("Layout Component when not signed in", () => {
  beforeEach(() => {
    render(<Home />)
  })
  it("Header element exists", () => {
    const header = screen.getByTestId("header-div")
    expect(header).toBeInTheDocument()
  })
  it("Footer element exists", () => {
    const footer = screen.getByTestId("footer-div")
    expect(footer).toBeInTheDocument()
  })
  it("Main element exists", () => {
    const main = screen.getByTestId("main-div")
    expect(main).toBeInTheDocument()
  })
  it("Header title exists", () => {
    const headerTitle = screen.getByTestId("header-title")
    expect(headerTitle).toBeInTheDocument()
  })
  it("Footer has Contact link", () => {
    const contactLink = screen.getByTestId("contact-link")
    expect(contactLink).toBeInTheDocument()
  })
  it("Footer has Sign In link", () => {
    const signInLink = screen.getByTestId("sign-in-link")
    expect(signInLink).toBeInTheDocument()
  })
})

/* describe("Layout when signed in", () => {
  it("Sign out link exists", () => {})
  it("Sign out link calls signout function", () => {})
   it("Menu icon exists", () => {
    const menuIcon = screen.getByTestId("menu-icon")
    expect(menuIcon).toBeInTheDocument()
  })
  //it("Menu icon opens and closes menu", () => {})
  //it("Menu has links to About, Contact and Syllabus", () => {})
}) */

describe("Home Component", () => {
  beforeEach(() => {
    render(<HomeComponent />)
  })
  it("Matches snapshot", () => {
    const homeComponent = render(<Home />)
    expect(homeComponent).toMatchSnapshot()
  })
  it("Has a welcome blurb", () => {
    const homeBlurb = screen.getByTestId("home-blurb")
    expect(homeBlurb).toBeInTheDocument()
  })
  it("Interactive piano keyboard in the component", () => {
    const pianoComponent = screen.getByTestId("piano-div")
    expect(pianoComponent).toBeInTheDocument()
  })
  it("About link in the component", () => {
    const aboutLink = screen.getByTestId("about-link")
    expect(aboutLink).toBeInTheDocument()
  })
  it("Exercises link in the component", () => {
    const exercisesLink = screen.getByTestId("exercises-link")
    expect(exercisesLink).toBeInTheDocument()
  })
})

/*
describe("About Component", () => {})
describe("Contact Component", () => {})
describe("Profile Component", () => {})
describe("Piano keyboard component", () => {
  it("Playing a note calls the corresponding tone", () => {})
  it("Toggling the alphabet button shows corresponding note names", () => {})
})

describe("Exercises Categories List", () => {
  it("Has a title element", () => {})
  it("Renders all categories of exercises", () => {})
  it("Category tiles have title and blurb", () => {})
})
describe("Exercise List", () => {
  it("Has a title element", () => {})
  it("Renders all categories of exercises", () => {})
  it("Category tiles have title and blurb", () => {})
})

 */