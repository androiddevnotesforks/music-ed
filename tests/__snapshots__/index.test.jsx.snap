// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col items-center w-screen "
        data-testid="home-component"
      >
        <div
          class="home-blurb"
          data-testid="home-blurb"
        >
          <h1
            class="p-2"
          >
            Welcome to the Aural Gymnasium.
          </h1>
          <p>
            This is a completely free resource for musicians to develop their relative pitch and chord recognition.
          </p>
          <p>
            You can use the exercises as much as you like. To track your progress, create an account.
          </p>
        </div>
        <div
          class="piano"
          data-testid="piano-div"
        >
          <button
            class="piano-btn"
            data-testid="note-name-toggle"
          >
            ABC
          </button>
          <div
            class="piano-keyboard"
          >
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="black-key text-sm font-thin active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
            <button
              class="text-sm font-thin white-key active:bg-red-400"
            />
          </div>
        </div>
        <div
          class="home-blurb"
        >
          <p>
            The Aural Gymnasium has over 40 exercises for musicians and students of any level to improve their aural skills. Answers are entered using the interactive keyboard.
          </p>
          <div
            class="flex flex-col items-center w-full "
          >
            <a
              href="/about"
            >
              <button
                class="text-green-500 border-green-500 hover:bg-green-50 active:bg-white rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
                data-testid="about-link"
              >
                About
              </button>
            </a>
            <a
              href="/exercises"
            >
              <button
                class=" text-blue-500 border-blue-500 hover:bg-blue-50 active:bg-white rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
                data-testid="exercises-link"
              >
                Get Started
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="layout "
      >
        <div
          class="header"
          data-testid="header-div"
        >
          <a
            href="/"
          >
            <h1
              class="header-link"
              data-testid="header-title"
            >
              Aural Gym
            </h1>
          </a>
        </div>
        <div
          class="flex flex-col w-screen min-h-screen p-2 items-center"
          data-testid="main-div"
        >
          <div
            class="flex flex-col items-center w-screen "
            data-testid="home-component"
          >
            <div
              class="home-blurb"
              data-testid="home-blurb"
            >
              <h1
                class="p-2"
              >
                Welcome to the Aural Gymnasium.
              </h1>
              <p>
                This is a completely free resource for musicians to develop their relative pitch and chord recognition.
              </p>
              <p>
                You can use the exercises as much as you like. To track your progress, create an account.
              </p>
            </div>
            <div
              class="piano"
              data-testid="piano-div"
            >
              <button
                class="piano-btn"
                data-testid="note-name-toggle"
              >
                ABC
              </button>
              <div
                class="piano-keyboard"
              >
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="black-key text-sm font-thin active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
                <button
                  class="text-sm font-thin white-key active:bg-red-400"
                />
              </div>
            </div>
            <div
              class="home-blurb"
            >
              <p>
                The Aural Gymnasium has over 40 exercises for musicians and students of any level to improve their aural skills. Answers are entered using the interactive keyboard.
              </p>
              <div
                class="flex flex-col items-center w-full "
              >
                <a
                  href="/about"
                >
                  <button
                    class="text-green-500 border-green-500 hover:bg-green-50 active:bg-white rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
                    data-testid="about-link"
                  >
                    About
                  </button>
                </a>
                <a
                  href="/exercises"
                >
                  <button
                    class=" text-blue-500 border-blue-500 hover:bg-blue-50 active:bg-white rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
                    data-testid="exercises-link"
                  >
                    Get Started
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="footer"
          data-testid="footer-div"
        >
          <a
            href="/contact"
          >
            <p
              class="p-2 text-blue-500 hover:underline hover:cursor-pointer"
              data-testid="contact-link"
            >
              Contact Us
            </p>
          </a>
          <button
            class="signin-info"
            data-testid="sign-in-link"
          >
            <div
              class="user-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3zm9 11v-1a7 7 0 0 0-7-7h-4a7 7 0 0 0-7 7v1h2v-1a5 5 0 0 1 5-5h4a5 5 0 0 1 5 5v1z"
                />
              </svg>
            </div>
            <p>
              Sign in
            </p>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout "
    >
      <div
        class="header"
        data-testid="header-div"
      >
        <a
          href="/"
        >
          <h1
            class="header-link"
            data-testid="header-title"
          >
            Aural Gym
          </h1>
        </a>
      </div>
      <div
        class="flex flex-col w-screen min-h-screen p-2 items-center"
        data-testid="main-div"
      >
        <div
          class="flex flex-col items-center w-screen "
          data-testid="home-component"
        >
          <div
            class="home-blurb"
            data-testid="home-blurb"
          >
            <h1
              class="p-2"
            >
              Welcome to the Aural Gymnasium.
            </h1>
            <p>
              This is a completely free resource for musicians to develop their relative pitch and chord recognition.
            </p>
            <p>
              You can use the exercises as much as you like. To track your progress, create an account.
            </p>
          </div>
          <div
            class="piano"
            data-testid="piano-div"
          >
            <button
              class="piano-btn"
              data-testid="note-name-toggle"
            >
              ABC
            </button>
            <div
              class="piano-keyboard"
            >
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="black-key text-sm font-thin active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
              <button
                class="text-sm font-thin white-key active:bg-red-400"
              />
            </div>
          </div>
          <div
            class="home-blurb"
          >
            <p>
              The Aural Gymnasium has over 40 exercises for musicians and students of any level to improve their aural skills. Answers are entered using the interactive keyboard.
            </p>
            <div
              class="flex flex-col items-center w-full "
            >
              <a
                href="/about"
              >
                <button
                  class="text-green-500 border-green-500 hover:bg-green-50 active:bg-white rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
                  data-testid="about-link"
                >
                  About
                </button>
              </a>
              <a
                href="/exercises"
              >
                <button
                  class=" text-blue-500 border-blue-500 hover:bg-blue-50 active:bg-white rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
                  data-testid="exercises-link"
                >
                  Get Started
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
        data-testid="footer-div"
      >
        <a
          href="/contact"
        >
          <p
            class="p-2 text-blue-500 hover:underline hover:cursor-pointer"
            data-testid="contact-link"
          >
            Contact Us
          </p>
        </a>
        <button
          class="signin-info"
          data-testid="sign-in-link"
        >
          <div
            class="user-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3zm9 11v-1a7 7 0 0 0-7-7h-4a7 7 0 0 0-7 7v1h2v-1a5 5 0 0 1 5-5h4a5 5 0 0 1 5 5v1z"
              />
            </svg>
          </div>
          <p>
            Sign in
          </p>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
